libnetwork/resolvconf/resolvconf_unix_test.go

```diff
@@ -416,11 +435,788 @@ func TestFilterResolvDNS(t *testing.T) {
+# ExtServers: [host(127.0.0.53)]
+# Overrides: []
+# Option ndots from: host
+`,
+		},
+		{
+			name:          "ndots missing but required",
+			input:         "nameserver 127.0.0.53",
+			reqdOptions:   []string{"ndots:0"},
+			expExtServers: []ExtDNSEntry{mke("127.0.0.53", true)},
+			expContent: `nameserver 127.0.0.11
+options ndots:0
+
+# Based on host file: '/etc/resolv.conf' (internal resolver)
+# ExtServers: [host(127.0.0.53)]
+# Overrides: []
+# Option ndots from: internal
+`,
+		},
+		{
+			name:            "ndots host, override and required",
+			input:           "nameserver 127.0.0.53",
+			reqdOptions:     []string{"ndots:0"},
+			overrideOptions: []string{"ndots:2"},
+			expExtServers:   []ExtDNSEntry{mke("127.0.0.53", true)},
+			expContent: `nameserver 127.0.0.11
+options ndots:2
+
+# Based on host file: '/etc/resolv.conf' (internal resolver)
+# ExtServers: [host(127.0.0.53)]
+# Overrides: [options]
+# Option ndots from: override
+`,
+		},
+		{
+			name:          "Extra required options",
+			input:         "nameserver 127.0.0.53\noptions trust-ad",
+			reqdOptions:   []string{"ndots:0", "attempts:3", "edns0", "trust-ad"},
+			expExtServers: []ExtDNSEntry{mke("127.0.0.53", true)},
+			expContent: `nameserver 127.0.0.11
+options trust-ad ndots:0 attempts:3 edns0
+
+# Based on host file: '/etc/resolv.conf' (internal resolver)
+# ExtServers: [host(127.0.0.53)]
+# Overrides: []
+# Option ndots from: internal
+`,
+		},
+		{
+			name:          "Bad internal nameserver",
+			input:         "nameserver 127.0.0.53",
+			intNameServer: "1.2.3.4.5",
+			expErr:        "ParseAddr(\"1.2.3.4.5\"): IPv4 address too long",
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc := tc
+			rc, err := readResolvConf("/etc/resolv.conf", bytes.NewBuffer([]byte(tc.input)))
+			assert.NilError(t, err)
+
+			if tc.intNameServer == "" {
+				tc.intNameServer = "127.0.0.11"
+			}
+			if len(tc.overrideNS) > 0 {
+				rc.OverrideNameServers(tc.overrideNS)
+			}
+			if len(tc.overrideOptions) > 0 {
+				rc.OverrideOptions(tc.overrideOptions)
+			}
+			err = rc.TransformForIntNS(tc.ipv6, tc.intNameServer, tc.reqdOptions)
+			if tc.expErr != "" {
+				assert.Check(t, is.ErrorContains(err, tc.expErr))
+				return
+			}
+			assert.NilError(t, err)
+
+			d := t.TempDir()
+			path := filepath.Join(d, "resolv.conf")
+			err = rc.WriteFile(path, "", 0644)
+			assert.NilError(t, err)
+
+			content, err := os.ReadFile(path)
+			assert.NilError(t, err)
+			// fmt.Print(string(content))
+			assert.Check(t, is.DeepEqual(string(content), tc.expContent))
+			assert.Check(t, is.DeepEqual(rc.ExtNameServers(), tc.expExtServers,
+				cmp.Comparer(func(a, b ExtDNSEntry) bool { return a == b })))
+		})
+	}
+}
+
+func TestRCRead(t *testing.T) {
+	d := t.TempDir()
+	path := filepath.Join(d, "resolv.conf")
+
+	// Try to read a nonexistent file, equivalent to an empty file.
+	rc, err := ReadResolvConfFile(path)
+	assert.Check(t, is.Equal(err, nil))
```

Comment:
```suggestion
	assert.Check(t, err)
```

Reply:
Now the decision to treat a missing file as empty is made outside the package, it's ...

`assert.Check(t, is.ErrorIs(err, fs.ErrNotExist))`